From 9f959af16de241eb63b686a1d0f1fed7c0d6cf8d Mon Sep 17 00:00:00 2001
From: Gyuyoung Kim <gyuyoung@igalia.com>
Date: Mon, 25 Feb 2019 04:12:25 +0000
Subject: [PATCH] BlinkMemoryMgt: Annotate MediaKeysGetStatusForPolicy and
 CopylessPasteExtractor with STATIC_ONLY

MediaKeysGetStatusForPolicy and CopylessPasteExtractor classes only have a static function.
So it makes sense to annonate STATIC_ONLY.

Bug: 919389
Change-Id: I4e1ec2b1e7815b4ac6eed1820b3800e2c30b6c78
Reviewed-on: https://chromium-review.googlesource.com/c/1484884
Reviewed-by: Kentaro Hara <haraken@chromium.org>
Commit-Queue: Gyuyoung Kim <gyuyoung@igalia.com>
Cr-Commit-Position: refs/heads/master@{#635028}
---
 .../modules/document_metadata/copyless_paste_extractor.h       | 3 +++
 .../modules/encryptedmedia/media_keys_get_status_for_policy.h  | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/third_party/blink/renderer/modules/document_metadata/copyless_paste_extractor.h b/third_party/blink/renderer/modules/document_metadata/copyless_paste_extractor.h
index b5954b0b268ae..301c33918ff2a 100644
--- a/third_party/blink/renderer/modules/document_metadata/copyless_paste_extractor.h
+++ b/third_party/blink/renderer/modules/document_metadata/copyless_paste_extractor.h
@@ -7,6 +7,7 @@
 
 #include "third_party/blink/public/platform/modules/document_metadata/copyless_paste.mojom-blink.h"
 #include "third_party/blink/renderer/modules/modules_export.h"
+#include "third_party/blink/renderer/platform/wtf/allocator.h"
 
 namespace blink {
 
@@ -16,6 +17,8 @@ class Document;
 // Copyless Paste feature. The extraction must be done after the document
 // has finished parsing.
 class MODULES_EXPORT CopylessPasteExtractor final {
+  STATIC_ONLY(CopylessPasteExtractor);
+
  public:
   static mojom::document_metadata::blink::WebPagePtr extract(const Document&);
 };
diff --git a/third_party/blink/renderer/modules/encryptedmedia/media_keys_get_status_for_policy.h b/third_party/blink/renderer/modules/encryptedmedia/media_keys_get_status_for_policy.h
index 57b1e568b0d41..81156ff29e3f6 100644
--- a/third_party/blink/renderer/modules/encryptedmedia/media_keys_get_status_for_policy.h
+++ b/third_party/blink/renderer/modules/encryptedmedia/media_keys_get_status_for_policy.h
@@ -6,6 +6,7 @@
 #define THIRD_PARTY_BLINK_RENDERER_MODULES_ENCRYPTEDMEDIA_MEDIA_KEYS_GET_STATUS_FOR_POLICY_H_
 
 #include "third_party/blink/renderer/bindings/core/v8/script_promise.h"
+#include "third_party/blink/renderer/platform/wtf/allocator.h"
 
 namespace blink {
 
@@ -14,6 +15,8 @@ class MediaKeysPolicy;
 class ScriptState;
 
 class MediaKeysGetStatusForPolicy {
+  STATIC_ONLY(MediaKeysGetStatusForPolicy);
+
  public:
   static ScriptPromise getStatusForPolicy(ScriptState*,
                                           MediaKeys&,
